src/Compilers/Core/Portable/Operations/ControlFlowGraph.cs

```diff
@@ -35,5 +46,48 @@ internal ControlFlowGraph(ImmutableArray<BasicBlock> blocks, Region root)
         /// Root (<see cref="RegionKind.Root"/>) region for the graph.
         /// </summary>
         public Region Root { get; }
+
+        /// <summary>
+        /// PROTOTYPE(dataflow): Add documentation
+        /// </summary>
+        public ImmutableArray<IMethodSymbol> Methods { get; }
+
+        /// <summary>
+        /// PROTOTYPE(dataflow): Add documentation
+        /// </summary>
+        public ControlFlowGraph this[IMethodSymbol method]
```

Comment:
> Is there some assert or check we can do to ensure that this argument is only ever a local function symbol? Same with the constructor parameter.

This is a public API. Why would we assert anything about its input?

Reply:
Misread the method. Same question about the constructor though.

---
In reply to: [191501882](https://github.com/dotnet/roslyn/pull/27216#discussion_r191501882) [](ancestors = 191501882)

Reply:
We should probably still throw an `ArgumentNullException` on null method, and an `ArgumentException` for a method which is not a key in `_methodsMap`. Can be done as a follow-up change.

---
In reply to: [191501882](https://github.com/dotnet/roslyn/pull/27216#discussion_r191501882) [](ancestors = 191501882)

Reply:
For now. Seems like a fairly simple change that will ensure our invariants are respected in future code changes.

---
In reply to: [191504875](https://github.com/dotnet/roslyn/pull/27216#discussion_r191504875) [](ancestors = 191504875)

Reply:
> We should probably still throw an ArgumentNullException on null method, and an ArgumentException for a method which is not a key in _methodsMap. Can be done as a follow-up change.

The indexing will throw on all invalid inputs, but I will add a prototype comment in case we want to throw our own exceptions instead.

Reply:
True the implementation of this method, as authored today, throws the correct exceptions. That's a very subtle implementation detail though. I had to look into the indexer on `ImmutableDictionary` to verify it properly throws `ArgumentNullException` in this case. A small change to the code path here could affect the exception being thrown. If we want to rely on this implicit behavior then I think we need a test to verify. Otherwise future developers could break this with a very small change to the method, 

---
In reply to: [191505569](https://github.com/dotnet/roslyn/pull/27216#discussion_r191505569) [](ancestors = 191505569)

Reply:
If we do add our own exceptions, we should also add corresponding unit tests for invalid inputs.

---
In reply to: [191505569](https://github.com/dotnet/roslyn/pull/27216#discussion_r191505569) [](ancestors = 191505569)

Reply:
> If we do add our own exceptions, we should also add corresponding unit tests for invalid inputs.

I am not planning to do that in context of this PR. I am comfortable with indexing catching bad things